### PR TITLE
Refactor ibm_pi_placement_group resource

### PR DESCRIPTION
**Pull request by ismirlia**
This PR refactors the ibm_pi_placement_group resource and cleans up the code.

```
--- PASS: TestAccIBMPIPlacementGroupBasic (3601.37s)
PASS
```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 1]
```diff
@@ -944,12 +953,76 @@ func isPIInstanceRefreshFunc(client *st.IBMPIInstanceClient, id, instanceReadySt
...
+
+	queryTimeOut := activeTimeOut
+
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Adding},
+		Target:     []string{State_Added},
+		Refresh:    isPIInstancePlacementGroupAddRefreshFunc(client, pgID, id),
+		Delay:      30 * time.Second,
+		MinTimeout: queryTimeOut,
+		Timeout:    120 * time.Minute,
```
Comment (yussufsh) [inline ibm/service/power/resource_ibm_pi_instance.go:967]: Don't you think 2 hrs is too much for adding/deleting PG?

Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_instance.go:967]: Yea that's a good point. Should lower that to like 10 minutes.

**Comment by ismirlia** [comment]
@yussufsh I fixed the merge conflicts.

**Review by yussufsh — APPROVED** [review]
lgtm

**Pull request merged**